code/http/url/url.php

```diff
@@ -612,18 +624,18 @@ public function toString($parts = self::FULL, $escape = null)
             //Add the username and password
             if (($parts & self::USER) && !empty($this->user))
             {
-                $url .= urlencode($this->user);
+                $url .= rawurlencode($this->user);
```

Comment:
@oligriffiths User is always set through setUser() which decoded on input and to be safe we encode on output to string. I don't see how that could lead to double encoding?